src/services/contactService.ts

```diff
@@ -1,53 +1,153 @@
 import {RemoveContactArgs} from '@sphereon/ssi-sdk.contact-manager';
-import {Party, Identity} from '@sphereon/ssi-sdk.data-store';
-import Debug, {Debugger} from 'debug';
-
-import {APP_ID} from '../@config/constants';
-import {cmAddContact, cmAddIdentity, cmGetContacts, cmRemoveContact, cmUpdateContact} from '../agent';
+import {Identity, Party} from '@sphereon/ssi-sdk.data-store';
+import {ActionType, EventLogger, EventLoggerBuilder, InitiatorType, LoggingEventType, LogLevel, SubSystem, System} from '@sphereon/ssi-sdk.core';
+import {agentContext, cmAddContact, cmAddIdentity, cmGetContacts, cmRemoveContact, cmUpdateContact} from '../agent';
 import {IAddIdentityArgs, ICreateContactArgs, IGetContactsArgs, IUpdateContactArgs} from '../types';
 
-const debug: Debugger = Debug(`${APP_ID}:contactService`);
+const logger: EventLogger = new EventLoggerBuilder()
+  .withContext(agentContext)
+  .withLogLevel(LogLevel.INFO)
+  .withSystem(System.CONTACT)
+  .withSubSystem(SubSystem.CONTACT_MANAGER)
+  .withInitiatorType(InitiatorType.SYSTEM)
+  .build();
 
 export const getContacts = async (args?: IGetContactsArgs): Promise<Array<Party>> => {
-  debug(`getContacts(${JSON.stringify(args)})...`);
+  await logger.logEvent({
```

Comment:
This is an interesting one. In some cases this is user initiated. Then we really need to have these as audit events and at the INFO level. In a lot of cases this could also be system initiated, simply to get available contacts for a UI for instance. In that case it would be like it is currently (TRACE and system initiated)

Reply:
The simple approach would be to add an additional audit trial argument to the method, but lets revisit when we actually have authz in place